flake.nix

```diff
@@ -31,6 +30,30 @@
             })
           ] ++ lib.optionals (stdenv.isLinux) [libudev-zero];
         };
+        stellarcli = stellardev // {
+          cargoLock = {
+            lockFile = ./Cargo.lock;
+          };
+
+          cargoLock.outputHashes = {
+              # The only git+https dependency in Cargo.lock
+              "stellar-rpc-client-21.4.0" = "sha256-ue7Ynux9YaDP3f/XkHz2OPd2g0iCX5R0yS5SaVHEYxQ";
+          };
+
+          # As of writing 'cargo test' fails
+          doCheck = false;
```

Comment:
Let's fix it separate to this PR so this isn't required.

Reply:
I think the idea is to fix tests first and then merge this PR with `doCheck = false;` removed, is it correct?

Reply:
It doesn't need to block this PR assuming nothing is actually broken, but once it is fixed we should circle back and remove the doCheck = false. I'm also a little confused as to why it's broken. The tests appear to pass for me outside of nix. Are they just failing only within nix?

Reply:
I actually think we should keep `doCheck = false;`. Our current tests are doing too much at the moment and have some dependencies (e.g. when you simply run tests they will fail, unless specifics steps are done) -- this makes it harder for is to maintain the flake. Second, this tests are unstable. If someone want to just try out the cli and uses `nix run` I think it'll be quite frustrating to not be able to run the cli (it already takes a lot of time to compile everything and it also takes a lot of time to run all the tests -- which should be overkill as we _should_ have al itl tested in our CI/Ci)
BTW, I also have tests failing in `nix run` but it works just fine in `nix develop -> cargo test`

@leighmcculloch thoughts?

Reply:
> Oh it works for you?

No sorry, I'm not using nix. I just meant the tests work when I run `cargo test`.



> I actually think we should keep `doCheck = false;`. Our current tests are doing too much

👍🏻

Reply:
@TomMD I think we are on agreement that we should revert latest changes made for `doCheck = true;` support, I think the revision before enabling tests was perfect